Umee-Bots/Large-Amount-Account-Involvement/src/agent.spec.ts

```diff
@@ -0,0 +1,131 @@
+    description: "Transaction includes large amount of addresses",
+    type: FindingType.Info,
+    severity: FindingSeverity.Info,
+    protocol: "Umee",
+    metadata: {
+      from,
+      to: to || "",
+      monitoredAddresses: JSON.stringify(addresses),
+      amountOfInvolvedAddresses: amount.toString(),
+    },
+  });
+};
+
+describe("Detect transaction involving large amount of addresses", () => {
+  let handleTransaction: HandleTransaction;
+
+  beforeAll(() => {
+    handleTransaction = provideHandleTransaction(CONFIG);
+  });
+
+  it("returns empty findings when there is not any transaction addresses involved besides 'from address'", async () => {
+    const tx = new TestTransactionEvent();
+
+    const findings = await handleTransaction(tx);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns empty findings when there are a large number of transaction addresses not involved with monitored addresses", async () => {
+    const tx = new TestTransactionEvent().addInvolvedAddresses(
+      createAddress("0x444"),
+      createAddress("0x555"),
+      ...testAddresses
+    );
+
+    const findings = await handleTransaction(tx);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns empty findings when there are not a large number of transaction addresses involved with monitored addresses", async () => {
+    const tx = new TestTransactionEvent().addInvolvedAddresses(
+      CONFIG.monitoredAddresses[0],
+      CONFIG.monitoredAddresses[1],
+      createAddress("0x666"),
+      createAddress("0x777")
+    );
+
+    const findings = await handleTransaction(tx);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("returns a finding when there are a large number of transaction addresses involved with a monitored address", async () => {
+    const tx = new TestTransactionEvent().addInvolvedAddresses(CONFIG.monitoredAddresses[0], ...testAddresses);
+
+    const amountOfInvolvedAddresses = Object.keys(tx.addresses).length;
+
+    const findings = await handleTransaction(tx);
+
+    expect(findings).toStrictEqual([
+      createFinding(tx.transaction.from, tx.transaction.to, [CONFIG.monitoredAddresses[0]], amountOfInvolvedAddresses),
+    ]);
+  });
+
+  it("returns a finding with two monitored addresses when there are a large number of transaction addresses involved with two monitored addresses in the same transaction", async () => {
+    const tx = new TestTransactionEvent().addInvolvedAddresses(
+      CONFIG.monitoredAddresses[0],
+      CONFIG.monitoredAddresses[1],
+      ...testAddresses
+    );
+
+    const amountOfInvolvedAddresses = Object.keys(tx.addresses).length;
+
+    const findings = await handleTransaction(tx);
+
+    expect(findings).toStrictEqual([
+      createFinding(
+        tx.transaction.from,
+        tx.transaction.to,
+        [CONFIG.monitoredAddresses[0], CONFIG.monitoredAddresses[1]],
+        amountOfInvolvedAddresses
+      ),
+    ]);
+  });
```

Comment:
Isn't this test basically equivalent to the next one?

Reply:
they're pretty similar. the next one just includes different addresses rather than only monitored ones.